autoload/SpaceVim/layers/lang/sml.vim

```diff
@@ -0,0 +1,124 @@
+"     repl_options = ''
+"     conceal = 1
+"     conceal_show_tick = 1
+"     auto_create_def_use = 'always'
+" <
+
+function! SpaceVim#layers#lang#sml#plugins() abort
+  let l:plugins = []
+  call add(l:plugins, ['jez/vim-better-sml', { 'on_ft' : 'sml', 'build' : 'make' }])
+  return l:plugins
+endfunction
+
+function! SpaceVim#layers#lang#sml#config() abort
+  call SpaceVim#layers#edit#add_ft_head_tamplate('sml', s:sml_file_head)
+  augroup spacevim_layer_lang_sml
+    autocmd!
+    " autocmd FileType sml setlocal omnifunc=SpaceVim#plugins#bashcomplete#omnicomplete
+    if s:sml_enable_conceal
+        autocmd FileType sml setlocal conceallevel=2
+    endif
+  augroup END
+  call SpaceVim#mapping#gd#add('sml', function('bettersml#jumptodef#JumpToDef'))
+  call SpaceVim#mapping#space#regesit_lang_mappings('sml', function('s:language_specified_mappings'))
+
+  let l:runner = {
+        \ 'exe' : g:sml_smlnj_executable,
+        \ 'opt' : [],
+        \ 'usestdin' : 1,
+        \ }
+  call SpaceVim#plugins#runner#reg_runner('sml', l:runner)
+  call SpaceVim#plugins#repl#reg('sml', g:sml_smlnj_executable . s:sml_repl_options)
+endfunction
+
+function! s:language_specified_mappings() abort
+  nnoremap <silent><buffer> K :call bettersml#typequery#TypeQuery()<CR>
+
+  call SpaceVim#mapping#space#langSPC('nmap', ['l','r'],
+        \ 'call SpaceVim#plugins#runner#open()',
+        \ 'execute current file', 1)
+
+  let g:_spacevim_mappings_space.l.s = {'name' : '+Send'}
+  call SpaceVim#mapping#space#langSPC('nmap', ['l','s', 'i'],
+        \ "call SpaceVim#plugins#repl#start('sml')",
+        \ "start REPL process", 1)
+  call SpaceVim#mapping#space#langSPC('nmap', ['l','s', 'l'],
+        \ "call SpaceVim#plugins#repl#send('raw', getline('.') . ';')",
+        \ "send line and keep code buffer focused", 1)
+  call SpaceVim#mapping#space#langSPC('nmap', ['l','s', 'b'],
+        \ 'call SpaceVim#plugins#repl#send("raw", join(getline(1, "$"), "\n") . ";")',
+        \ 'send buffer and keep code buffer focused', 1)
+  call SpaceVim#mapping#space#langSPC('nmap', ['l','s', 's'],
+        \ 'call SpaceVim#plugins#repl#send("raw", join(getline("''<", "''>"), "\n") . ";")',
+        \ 'send selection and keep code buffer focused', 1)
+endfunction
+
+let s:sml_file_head = ['']
+let s:sml_repl_options = ''
+let s:sml_enable_conceal = 0
+
+function! SpaceVim#layers#lang#sml#set_variable(var) abort
+  let g:sml_smlnj_executable = get(a:var, 'smlnj_path', 'sml')
+  let g:sml_mlton_executable = get(a:var, 'mlton_path', 'mlton')
+  let s:sml_repl_options = get(a:var, 'repl_options', s:sml_repl_options)
+  let g:sml_repl_options = s:sml_repl_options
+  let g:sml_auto_create_def_use = get(a:var, 'auto_create_def_use', 'mlb')
+  let g:sml_greek_tyvar_show_tick = get(a:var, 'enable_conceal_show_tick', '0')
+  let s:sml_enable_conceal = get(a:var, 'enable_conceal', s:sml_enable_conceal)
+  let s:sml_file_head = get(a:var, 'file_head', s:sml_file_head)
```

Comment:
in `lang#lua` and `lang#python`, we use `lua_file_head` and `python_file_head` as the layer option name, I think maybe we should we `sml_file_head`.

Reply:
updated to ``sml_file_head``.
I found out that ``python`` layer use both ``python-file-head`` and ``python_file_head``, but ``ruby`` and ``sh`` only support the ``ruby-file-head`` and ``bash-file-head``, maybe we should correct them.

Reply:
we should use sml_file_head, python-file-head is just for backward compatibility.